target/linux/ath79/dts/ar9330_teltonika_rut230.dts

```diff
@@ -0,0 +1,200 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "ar9330.dtsi"
+
+/ {
+	compatible = "teltonika,rut230", "qca,ar9330";
+	model = "Teltonika RUT230";
+
+	chosen {
+		bootargs = "console=ttyATH0,115200 root=31:05 rootfstype=squashfs";
+	};
+
+
+	aliases {
+		serial0 = &uart;
+		led-boot = &signal3;
+		led-failsafe = &signal3;
+		led-upgrade = &signal3;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 22 GPIO_ACTIVE_HIGH>;
+			debounce-interval = <60>;
+		};
+
+		input {
+			label = "input";
+			linux,code = <BTN_0>;
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+		sim {
+			label = "SIM holder";
+			linux,code = <KEY_RFKILL>;
```

Comment:
This does not match the name/label. Is this vendor behavior?

Reply:
Yes, not sure what for, but it was hard coded in the vendor firmware

Reply:
Still, if it's called "SIM holder", it shouldn't do an RFKILL. So maybe it's better to replace the linux,code by BTN_1.
Despite, I do not find input/sim in the documentation:
https://wiki.teltonika-networks.com/view/RUT230_front_and_back_panels

Reply:
for sim its triggered to gpio when sim cart is inserted and apparently it reloads modem without rebooting it and for the input, it is in the power socket. 
https://wiki.teltonika-networks.com/view/RUT230_Input/Output

Reply:
However, RFKILL won't reboot, but just toggle on/off ...?

Reply:
Ok, i found something. rfkill is defined in /etc/rc.button/ on the vendor firmware:
#!/bin/sh

. /lib/functions.sh
modem=$(gsmctl -n -y | head -c 4)

logger "$BUTTON was $ACTION for $SEEN seconds"

if [ "${ACTION}" = "released" ]
then
	if [ "$modem" = "EC25" -o "$modem" = "UC20" ]
	then
		ifdown ppp
		sleep 2
		/etc/init.d/operctl stop
		gsmctl -A "AT+CFUN=0"
	else
		/etc/init.d/modem stop
	fi

	/etc/init.d/ledsman restart
elif [ "${ACTION}" = "pressed" ]
then
	if [ "$modem" = "EC25" -o "$modem" = "UC20" ]
	then
		gsmctl -A "AT+CFUN=1"
		sleep 1
		/etc/init.d/gsmd restart
	else
		sleep 1
		/etc/init.d/modem restart
	fi
fi 

So its just restarts some services and changes modem functionality. This just waits for state to change but the gpio button itself toggles on/off when inserting/removing sim card. I guess it won't hurt to change linux code itself then, because same thing can be added for BTN_1 in rc.button ?